hacep/src/main/java/org/kie/hacep/consumer/DroolsConsumerHandler.java

```diff
@@ -73,35 +96,36 @@ public void process( ItemToProcess item, State state) {
         RemoteCommand command  = deserialize((byte[])item.getObject());
 
         if(config.isUnderTest()) {  loggerForTest.warn("Remote command on process:{}", command); }
-
         if (state.equals(State.LEADER)) {
             processCommand( command, state );
             Queue<Object> sideEffectsResults = DroolsExecutor.getInstance().getAndReset();
-            if (config.isUnderTest()) { loggerForTest.warn("sideEffectOnLeader:{}", sideEffectsResults); }
-
             ControlMessage newControlMessage = new ControlMessage(command.getId(), sideEffectsResults);
             producer.produceSync(config.getControlTopicName(), command.getId(), newControlMessage);
+            if (config.isUnderTest()) { loggerForTest.warn("sideEffectOnLeader:{}", sideEffectsResults); }
         } else {
             processCommand( command, state );
         }
     }
 
     public void processSideEffectsOnReplica(Queue<Object> newSideEffects) {
-        if(config.isUnderTest()){ loggerForTest.warn("sideEffectOnReplica:{}", newSideEffects);}
         DroolsExecutor.getInstance().appendSideEffects(newSideEffects);
+        if(config.isUnderTest()){ loggerForTest.warn("sideEffectOnReplica:{}", newSideEffects);}
     }
 
     @Override
     public void processWithSnapshot(ItemToProcess item, State currentState) {
         if (logger.isInfoEnabled()){ logger.info("SNAPSHOT"); }
         process(item, currentState);
-        snapshooter.serialize(kieSessionContext, item.getKey(), item.getOffset());
+        if(((StatefulKnowledgeSessionImpl)kieSessionContext.getKieSession()).isAlive()) {
```

Comment:
Same as above, is the alive check necessary?

Reply:
No